docker/importer/importer.py

```diff
@@ -536,6 +655,23 @@ def process_updates(self, source_repo: osv.SourceRepository):
     logging.error('Invalid repo type: %s - %d', source_repo.name,
                   source_repo.type)
 
+  def process_deletions(self, source_repo: osv.SourceRepository):
+    """Process source record deletions by withdrawing them."""
+    if source_repo.type == osv.SourceRepositoryType.GIT:
+      # TODO: To be implemented.
+      return
+
+    if source_repo.type == osv.SourceRepositoryType.BUCKET:
+      self._process_deletions_bucket(source_repo)
+      return
+
+    if source_repo.type == osv.SourceRepositoryType.REST_ENDPOINT:
```

Comment:
We should add a TODO to refactor some of this code, before we even implement the missing support for REST/git.
 
Rather than have all these if statements for importing/deleting, we should start defining some SourceRetriever classes that define the source-type-specific ways to retrieve records.

i.e. something along the lines of:

```python

class SourceRetriver

  def list() -> list[str]:
    """Return list of vuln locators for the source."""

  def get(locator: str) -> Vulnerability:
    """Return a vulnerability for the given locator."""
```

This way we can consolidate the import and deletion logic, and handle the differences via this abstraction.

Reply:
TODO added in 5fc725a